packages/flutter_tools/lib/src/test/runner.dart

```diff
@@ -89,14 +124,59 @@ Future<int> runTests(
 
   // Call package:test's main method in the appropriate directory.
   final Directory saved = fs.currentDirectory;
+
+  void printReloadMessage() {
+    printStatus(
+      "Press 'r' to rerun your tests, 'q' to quit",
+      color: TerminalColor.red
+    );
+  }
+
   try {
     if (workDir != null) {
       printTrace('switching to directory $workDir to run tests');
       fs.currentDirectory = workDir;
     }
 
     printTrace('running test package with arguments: $testArgs');
-    await test.main(testArgs);
+    await test.runTests(testArgs);
+
+    if (watchTests) {
```

Comment:
Lets move this entirely to a _watchTests method that this one `await`s

Reply:
Done